src/views/product-root-docs-path-landing/server.ts

```diff
@@ -73,70 +88,108 @@ const generateHeadingLevelsAndSidecarHeadings = ({
-
-		// TODO: replace any with accurate type
-		const generatedProps = (await generatedGetStaticProps({
-			...context,
-			params: { page: [] },
-		})) as $TSFixMe
-
-		// Append headings found in marketing content
-		const { sidecarHeadings, marketingContentBlocksWithHeadingLevels } =
-			generateHeadingLevelsAndSidecarHeadings({
-				layoutHeadings: generatedProps.props.layoutProps.headings,
-				marketingContentBlocks: pageContent.marketingContentBlocks,
-				pageTitle: `${product.name} ${baseName}`,
-			})
-
-		// TODO clean this up so it's easier to understand
-		return {
-			...generatedProps,
-			props: {
-				...generatedProps.props,
-				mdxSource: includeMDXSource ? generatedProps.props.mdxSource : null,
-				layoutProps: {
-					...generatedProps.props.layoutProps,
-					githubFileUrl: null,
-					headings: sidecarHeadings,
-				},
-				pageContent: {
-					...pageContent,
-					marketingContentBlocks: marketingContentBlocksWithHeadingLevels,
-				},
-				pageHeading: sidecarHeadings[0],
-				product: {
-					...generatedProps.props.product,
-					currentRootDocsPath,
-				},
+	/**
+	 * Fetch page marketing content. Falls back to empty object if
+	 * `src/content/${product.slug}/docs-landing.json` file is not present.
+	 *
+	 * Note: could consider other content sources. For now, JSON.
+	 * Asana task: https://app.asana.com/0/1100423001970639/1201631159784193/f
+	 */
+	let CONTENT
+	try {
+		const jsonFilePath = path.join(
+			process.cwd(),
+			`src/content/${product.slug}/docs-landing.json`
+		)
+		CONTENT = JSON.parse(fs.readFileSync(jsonFilePath, 'utf8'))
+	} catch {
+		CONTENT = {}
+	}
+
+	const { getStaticProps: generatedGetStaticProps } =
+		_getStaticGenerationFunctions({
+			product,
+			productSlugForLoader,
+			basePath: BASE_PATH,
+			baseName,
+		})
+
+	// TODO: replace any with accurate type
+	const generatedProps = (await generatedGetStaticProps({
+		...context,
+		params: { page: [] },
+	})) as $TSFixMe
+
+	// Append headings found in marketing content
+	const { sidecarHeadings, marketingContentBlocksWithHeadingLevels } =
+		generateHeadingLevelsAndSidecarHeadings({
+			layoutHeadings: generatedProps.props.layoutProps.headings,
+			marketingContentBlocks: CONTENT.marketingContentBlocks,
+			pageTitle: `${product.name} ${baseName}`,
+		})
+
+	// TODO clean this up so it's easier to understand
+	return {
+		...generatedProps,
+		props: {
+			...generatedProps.props,
+			mdxSource: includeMDXSource ? generatedProps.props.mdxSource : null,
+			layoutProps: {
+				...generatedProps.props.layoutProps,
+				githubFileUrl: null,
+				headings: sidecarHeadings,
 			},
-		}
+			pageContent: {
+				...CONTENT,
+				marketingContentBlocks: marketingContentBlocksWithHeadingLevels,
+			},
+			pageHeading: sidecarHeadings[0],
+			product: {
+				...generatedProps.props.product,
+				currentRootDocsPath,
+			},
+		},
 	}
 }
 
-export { generateGetStaticProps }
+export { generateRootDocsLandingPaths, generateRootDocsLandingProps }
```

Comment:
🗒️ Am now exporting these helpers rather than generators for the static paths/props getters.